lib/node_modules/@stdlib/math/base/special/frexp/src/main.c

```diff
@@ -0,0 +1,80 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/frexp.h"
+#include "stdlib/number/float64/base/normalize.h"
+#include "stdlib/number/float64/base/exponent.h"
+#include "stdlib/number/float64/base/from_words.h"
+#include "stdlib/math/base/assert/is_infinite.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/number/float64/base/get_high_word.h"
+#include <stdint.h>
+
+// Exponent all 0s: 1 00000000000 11111111111111111111 => 2148532223
+static const int32_t CLEAR_EXP_MASK = 2148532223;
+
+// Exponent equal to 1022 (BIAS-1): 0 01111111110 00000000000000000000 => 1071644672
+static const int32_t SET_EXP_MASK = 1071644672;
+
+/**
+* Splits a double-precision floating-point number into a normalized fraction and an integer power of two.
+*
+* @private
+* @param x        input value
+* @param frac     Destination to store the normalized fraction
+* @param exp      Destination to store the exponent
+*
+* @example
+* #include <stdint.h>
+*
+* double x = 3.141592653589793;
+* int32_t exp;
+* double frac;
+*
+* stdlib_base_frexp( x, &frac, &exp );
+*/
+void stdlib_base_frexp( const double x, double *frac, int32_t* exp ) {
+	if ( x == 0.0 || stdlib_base_is_nan( x ) || stdlib_base_is_infinite( x ) ) {
+		*frac = x;
+		*exp = 0;
+		return;
+	}
+	double *y;
+	int32_t *e;
+
+	// Normalize the input value:
+	stdlib_base_float64_normalize( x, y, e );
+
+	// Extract the exponent from `x` and add the normalization exponent:
+	*exp = stdlib_base_float64_exponent( *y ) + *e + 1;
+
+	// Break `x` into two unsigned 32-bit integers (higher and lower order words):
+	uint32_t high;
+	stdlib_base_float64_get_high_word( x, &high );
```

Comment:
You need to use `@stdlib/number/base/to-words` here. L77 is not correct. We want to retain the low word of `x`.

Reply:
See the JavaScript implementation.

Reply:
Okay so what we are trying to do is break `x` into `high` and `low` and then have `x = fromWords( high, WORDS[ 1 ] );` (Copied from assign.js) right?

Reply:
And for additional reference: <https://github.com/freemint/fdlibm/blob/f0f03ccb6337caf663c24f64d062e6f3ff20d1df/s_frexp.c>. For some of the logic, we've extracted away to separate functions/packages, but this should serve as a decent reference.

Reply:
Yes, but the C `to_words` API does not use arrays. Need to use pointers.